Makefile

```diff
@@ -12,18 +12,17 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-ROOT_DIR_RELATIVE := .
+export REPO_ROOT := $(shell git rev-parse --show-toplevel)
```

Comment:
David made an argument about this being necessary as sometimes CAPC and other providers are included in other projects (Potentially CAPI itself). Using `ROOT_DIR_RELATIVE` is supposed to allow for this. I'm not sure we should remove this without fully understanding the implications.

Reply:
I think @davidjumani may be in a conference and unavailable. 

The CAPC repo is also checked out as a subdirectory in the EKS-A build-tooling repo: https://github.com/aws/eks-anywhere-build-tooling/tree/main/projects/kubernetes-sigs/cluster-api-provider-cloudstack. When we build it, we clone the CAPC repo and build it below with make -C. I feel like using git rev-parse is safer than hardcoding `.`, which becomes ambiguous if you're running the build from a different directory

Reply:
I think what David meant was that CAPC was used as a [submodule](https://www.git-scm.com/book/en/v2/Git-Tools-Submodules)

In that case the repo root according to git would no longer be the CAPC repo's root.

I'm not that familiar with submodules or David's reasoning/use case. So, if you think we really should remove this, then I suppose I'm fine with it. We've certainly released just fine without this change before. Of course most of our previous releases weren't actually functional from a `clusterctl init` standpoint.

Reply:
`REPO_ROOT` is already defined in https://github.com/kubernetes-sigs/cluster-api-provider-cloudstack/blob/main/Makefile#L26

`ROOT_DIR_RELATIVE` uses a similar pattern defined in other providers as well
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/Makefile#L15
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/Makefile#L15

I think this was needed for building the books, or something similar. Will get back to you on that.

Reply:
>REPO_ROOT is already defined in https://github.com/kubernetes-sigs/cluster-api-provider-cloudstack/blob/main/Makefile#L26, maybe it can be reused ?

This is where I got the repo root from. It's deleted in line 26 below

>Another reason was to align it with other providers. Eg:
ROOT_DIR_RELATIVE uses a similar pattern defined in other providers as well

I'm looking at CAPV and I see they have a fancy way to identify the root as well (https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/Makefile#L41) so I think using git repo root should be appropriate here, or maybe vsphere's way.